rust_src/src/lread.rs

```diff
@@ -155,6 +159,41 @@ pub unsafe fn defvar_per_buffer_offset(
     }
 }
 
+/// Read a byte from stdio. If it has lookahead, use the stored value.
+/// If read over network is interrupted, keep trying until read succeeds.
+#[no_mangle]
+pub unsafe extern "C" fn readbyte_from_stdio() -> i32 {
+    // If infile has lookahead, use stored value
+    if (*infile).lookahead > 0 {
+        let idx = (*infile.sub(1)).lookahead as usize;
```

Comment:
This does not have the same behavior. The C code is decrementing `lookahead` not `infile`. It should read:

    if (*infile).lookahead > 0 {
        (*infile).lookahead -= 1;
        return (*infile).buf[(*infile).lookahead as usize];
    }

Reply:
That makes way more sense, I thought the way I read it seemed off.

Reply:
Not having to do pointer math also means I can convert the pointer to a reference, which means I won't have to dereference it explicitly so many times.